### ISSUE TITLE
create a test runner

### ISSUE THREAD

**Issue by gabrielfalcao** [gabrielfalcao/sure#23]
sure should be a test runner and just like [lettuce](http://lettuce.it), provide the following feature:
## a "bootstrap" file

when running sure should try to import a `confidency.py` and there the developer will be able to declare some "hooks":

``` python
# confidence.py

from sure.hooks import before, after

@before.each_test
def clear_database(test):
    if "myapp.redis_layer" in test.__module__:
        # do whatever actions specific to tests under myapp.redis_layer


```

rant of possible hooks:
- `@before.each_test`
- `@before.each_module`
- `@before.each_package`
- `@before.all`
- `@after.each_test`
- `@after.each_module`
- `@after.each_package`
- `@after.all`

EDIT by @timofurrer: _content of former TODO.md_:

> ## Test runner with chronometer and coverage reports
> 
> _Sure_ provides you with a test runner that gives you full control
> over the process to the granularity of a single test case.
> 
> You can write your tests anywhere, _sure_ will find, run, count up the
> duration and allow running every test case, in the end you get
> acquainted with the statistics of your python codebase: what parts
> have test coverage, where you should refactor or add coverage.
> 
> It also runs [flake8](http://pypi.python.org/pypi/flake8/) against
> your test code and blame you when your tests are too complex
> 
> Ahh yes, you can also run only a given subset of test cases, force
> slow tests to fail.

**Comment by timofurrer** [comment]
I think this only makes sense if there is a big advantage over `pytest`, `nose`, `unittest` and other existing test runners.

Any more details on your idea?

**Comment by gabrielfalcao** [comment]
I agree, I personally have been using nose, exclusively, from the beginning.
I think that this should have the lowest priority for now :)

**Comment by timofurrer** [comment]
I totally agree. Let's keep the issue open though - I'll label it accordingly.

**Comment by gabrielfalcao** [comment]
Presently developing a runner, this is the highest priority for sure
